main.go

```diff
@@ -0,0 +1,334 @@
+package main
+
+import (
+    "bytes"
+
+    "fmt"
```

Comment:
you can remove most of these newlines. It is generally accepted to put a newline between stdlib packages and imported and external packages, though